articles/api-auth/client-types.md

```diff
@@ -0,0 +1,31 @@
+---
+title: Public and confidential clients
+---
+
+# Client and application types
+
+The OAuth 2.0 specification [defines two types of clients](https://tools.ietf.org/html/rfc6749#section-2.1): public and confidential.
+When creating a client through the management dashboard, Auth0 will ask you what type of application it represents and use that information to determine the client type.
+
+## Confidential clients
+
+Confidential clients are able to hold credentials, i.e. a client ID and secret, in a secure way without exposing it to users or attackers.
+This means that a trusted backend server is needed to store the secrets.
+The following application types are confidential clients:
+
+* Regular web applications using the [Authorization Code grant](/api-auth/grant/authorization-code)
```

Comment:
I would also add the two active grant types that are available via the `/oauth/token` endpoint (`password` and `password-realm`) to this list. This supports the use case of a first-party regular web application that doesn't need SSO, but collects user credentials directly. It posts those credentials to its own backend endpoint and then uses either the `password` and `password-realm` grant types to _securely_ authenticate the user with Auth0 using a `client_id` and `client_secret`.

This is related to this [Slack conversation](https://auth0.slack.com/archives/p2-rollout/p1488651715029121) in `#p2_rollout`.